packages/gatsby/src/bootstrap/load-plugins/load.js

```diff
@@ -243,6 +243,15 @@ const loadPlugins = (config = {}, rootDir = null) => {
     }
   }
 
+  // TypeScript support by default! use the user-provided one if it exists
+  const typescriptPlugin = (config?.plugins || []).find(
+    plugin => plugin.resolve === `gatsby-plugin-typescript`
+  ) || {
+    resolve: require.resolve(`gatsby-plugin-typescript`),
+  }
+
+  plugins.push(processPlugin(typescriptPlugin))
```

Comment:
Should we add it to the array again when the user has provided this already?

Reply:
At this stage we won't dedupe - we have some deduping logic implemented, but this is done when merging `gatsby-config` (themes) which is done before this code is executing

Reply:
this was a bug actually, fixed it